### PR TITLE
Combine eslint plugin and config packages to one

### LINKED ISSUE #31204: Bring ES Lint packages in Calypso up to date
New packages:
- https://github.com/Automattic/wp-calypso/tree/master/packages/eslint-config-wpcalypso
- https://github.com/Automattic/wp-calypso/tree/master/packages/eslint-plugin-wpcalypso

Old packages:
- https://github.com/Automattic/eslint-config-wpcalypso
- https://github.com/Automattic/eslint-plugin-wpcalypso

See also https://eslint.org/docs/user-guide/configuring#using-a-shareable-configuration-package for how to use these packages in Jetpack repo.

@Automattic/team-calypso

### PR DESCRIPTION
**Pull request by simison**
Combines `eslint-plugin-wpcalypso` and `eslint-config-wpcalypso` packages now that Eslint supports it:

https://eslint.org/docs/developer-guide/working-with-plugins#configs-in-plugins

### TODO
- [ ] Update readme
- [ ] Upgrading instructions for consumers 4.x → 5.x

#### Changes proposed in this Pull Request

* 
#### Testing instructions

—

Resolves https://github.com/Automattic/wp-calypso/issues/31204

### CODE REVIEW

File: packages/eslint-plugin-wpcalypso/package.json [hunk 1]
```diff
@@ -20,5 +20,12 @@
 	"engines": {
 		"node": ">=10"
 	},
+	"peerDependencies": {
+		"eslint": "^3.3.1 || ^4.6.1 || ^5.0.0",
+		"eslint-plugin-wpcalypso": "^3.4.1 || ^4.0.0"
```
Comment (sirreal) [inline packages/eslint-plugin-wpcalypso/package.json]: I don't think the module should depend on itself 🙂 

```suggestion
```

**Review by sirreal — COMMENTED** [review]
Awesome, thanks for taking this on!

File: packages/eslint-plugin-wpcalypso/package.json (hunk 1 above)
Comment (simison) [reply packages/eslint-plugin-wpcalypso/package.json]: Well spotted! 😆

**Comment by simison** [comment]
Relevant convo: https://github.com/Automattic/wp-calypso/pull/31902/files#r275643248

**Comment by sirreal** [comment]
This would still be nice to do, but hard to prioritize. Shall we close it?

**Comment by simison** [comment]
Yass.

cc @Automattic/team-calypso in case someone there wants to tackle this. :-)

**Comment by scinos** [comment]
I'll take care of this

**Comment by scinos** [comment]
Revived in #48310

**Pull request closed without merging**